padl/dumptools/serialize.py

```diff
@@ -44,8 +46,20 @@ def __init__(self, val: Any, save_function: Callable, load_function: callable,
     def save(self, path: Path):
         if path is None:
             path = Path('?')
-        filename = self.save_function(self.val, path, self.i)
-        complete_path = f"pathlib.Path(__file__).parent / '{filename}'"
+        if self.file_suffix is not None:
+            path = Path(str(path) + f'/{self.i}{self.file_suffix}')
+        filename = self.save_function(self.val, path)
+        if filename is None:
+            assert self.file_suffix is not None, ('if no file file_suffix is passed to *value*, '
+                                                  'the *save*-function must return a filename')
+            filename = path.name
+        if isinstance(filename, (str, Path)):
+            complete_path = f"pathlib.Path(__file__).parent / '{filename}'"
+        else:
```

Comment:
Instead of using `else` could we use an `isinstance(filename, List)` and raise an exception if a different type is provided? This could catch an error in a custom written `save_function`